### PR TITLE
Check for more bugs in CI to reduce the amount of regressions

### PR DESCRIPTION
**Pull request by qarmin**
Added more CI checks(with examples) to prevent regressions like - https://github.com/godotengine/godot/issues/51764 or 

Also fixes 
```
drivers/unix/file_access_unix.cpp:268:2: runtime error: null pointer passed as argument 1, which is declared to never be null
```
(needed to pass new CI) - https://github.com/godotengine/godot/runs/3365003107#step:12:14

~For now CI probably will fail due this errors(I added simple workaround, but probably this is not right fix)~
For now checking for vptr(error below) is disabled - this waiting for fix in https://github.com/godotengine/godot/issues/51888, but probably better is to do it in next PR.
```
 servers/rendering/renderer_scene_cull.cpp:1299:94: runtime error: downcast of address 0x607000c4d2a0 which does not point to an object of type 'InstanceGeometryData'
0x607000c4d2a0: note: object is of type 'RendererSceneCull::InstanceVisibilityNotifierData'
 be be be be  90 b5 c1 24 00 00 00 00  00 be be be be be be be  00 00 00 00 00 00 00 00  01 00 00 00
              ^~~~~~~~~~~~~~~~~~~~~~~
              vptr for 'RendererSceneCull::InstanceVisibilityNotifierData'
```

### CODE REVIEW

**Review by pouleyKetchoupp — APPROVED** [review]
The fix in physics looks fine.

File: misc/scripts/check_ci_log.py
Comment (akien-mga) [inline misc/scripts/check_ci_log.py:52]:
```suggestion
    print('ERROR: Memory leak was found (search for "ObjectDB instances leaked at exit" in CI log)')
```

Same ofor other errors, a space before a parenthesis is important for readability.

**Comment by fire** [comment]
@reduz Is this ready to be merged?

**Comment by fire** [comment]
The approval is blocked on the core team not following CI and requiring some time update understanding.

File: misc/scripts/check_ci_log.py
Comment (reduz) [inline misc/scripts/check_ci_log.py:117]: I think this is going to trigger some bugs as the rendering code is not entirely stable yet.

Comment (qarmin) [reply misc/scripts/check_ci_log.py:117]: I know that this is unstable so for now it only prints warning message and CI still works fine when this error appear.

**Review by mhilbrunner — COMMENTED** [review]
This looks nice, I left a few comments below.
This uses single quotes for strings, as opposed to the double quotes used elsewhere. Is there a reason for this? IMO this should use double quotes, like is standard in the code base.

**Comment by mhilbrunner** [comment]
@qarmin Friendly bump. :)

**Comment by qarmin** [comment]
Probably better would be create separate issue and PR for each bug-fix, so I will do it today or tomorrow.

**Pull request closed without merging**
